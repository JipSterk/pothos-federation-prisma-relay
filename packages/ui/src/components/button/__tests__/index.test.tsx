import { render } from "@testing-library/react";
import { Button } from "..";

describe("Button", () => {
  it("should be primary when not given an intent", () => {
    const { asFragment } = render(<Button />);
    const fragment = asFragment();

    const button = fragment.querySelector("button") as HTMLButtonElement;

    expect(button.classList.contains("bg-indigo-600"));
    expect(fragment).toMatchSnapshot();
  });

  it("should apply styling based on an intent", () => {
    const { asFragment } = render(<Button intent="secondary" />);
    const fragment = asFragment();

    const button = fragment.querySelector("button") as HTMLButtonElement;

    expect(button.classList.contains("bg-indigo-100"));
  });
});
