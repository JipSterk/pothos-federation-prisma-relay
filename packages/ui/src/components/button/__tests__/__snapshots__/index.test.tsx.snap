// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a button 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center rounded-md border border-transparent px-4 py-2 text-sm font-medium focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2 bg-indigo-600 text-white shadow-sm hover:bg-indigo-700"
    type="button"
  />
</DocumentFragment>
`;
